modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/IgniteTypeCoercion.java

```diff
@@ -47,6 +49,57 @@ public IgniteTypeCoercion(RelDataTypeFactory typeFactory, SqlValidator validator
         super(typeFactory, validator);
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean binaryComparisonCoercion(SqlCallBinding binding) {
+        // Although it is not reflected in the docs, this method is also invoked for MAX, MIN (and other similar operators)
+        // by ComparableOperandTypeChecker. When that is the case, fallback to default rules.
+        SqlCall call = binding.getCall();
+        if (binding.getOperandCount() != 2 && !SqlKind.BINARY_COMPARISON.contains(call.getKind())) {
```

Comment:
it there any binary comparison with operands count != 2?

Reply:
Discussed in slack. There was a problem since we should use OR instead of AND here.